app/scripts/directives/editEnvironmentFrom.js

```diff
@@ -0,0 +1,192 @@
+"use strict";
+(function() {
+  angular.module("openshiftConsole").component('editEnvironmentFrom', {
+    controller: [
+      '$attrs',
+      '$filter',
+      'keyValueEditorUtils',
+      EditEnvironmentFrom
+    ],
+    bindings: {
+      addRowLink: '@',
+      entries: '=',
```

Comment:
I'm not sure I'm crazy about `envFromSelectorOptions`...I'd at least like a comment about what it means.  Its config for something, but at a glance I'm not sure what (might be my ignorance re: EnvironmentFrom).

Reply:
An annotation couldn't hurt, we'll add one in